pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -79,8 +80,8 @@ func setAccessLog(env *model.Environment, node *model.Proxy, config *tcp_proxy.T
 
 // setAccessLogAndBuildTCPFilter sets the AccessLog configuration in the given
 // TcpProxy instance and builds a TCP filter out of it.
-func setAccessLogAndBuildTCPFilter(env *model.Environment, node *model.Proxy, config *tcp_proxy.TcpProxy) *listener.Filter {
-	setAccessLog(env, node, config)
+func setAccessLogAndBuildTCPFilter(meshConfig *meshconfig.MeshConfig, node *model.Proxy, config *tcp_proxy.TcpProxy) *listener.Filter {
```

Comment:
@rshriram  I don't think this interface change is needed or good - passing env to all methods is better than passing just the mesh config ( even if methods happen to only use parts of env)

Reply:
So there were a couple of reasons that I chose to pass the mesh config around:

1. avoiding the (minor) overhead for the atomic read every time the mesh config is needed.
2. ensuring that we have a consistent view of the mesh config throughout the entire xds push. Before, it would be possible for the mesh config to change in the middle of a config push, which I suspect would be undesirable.

It's not clear to me what parts of the mesh config are allowed to be changed externally. It would be a lot simpler if the mesh config were just constant and we never had to worry about this.  In that case, just passing around Env would be a lot simpler.

Thoughts?